SqlAssignments.sql

```diff
@@ -0,0 +1,1846 @@
+| Serres | Liu      | 13   |
+| Serres | Grass    | 13   |
+| Rifkin | Cisneros | 15   |
++--------+----------+------+
+3 rows in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+
+
+Que 26 -> Find salespeople who have multiple customers.
+
+Query  ->SELECT snum, COUNT(snum) AS total_customers  
+       ->FROM Customers  
+       ->GROUP BY snum  
+       ->HAVING COUNT(snum)>1;
++------+-----------------+
+| snum | total_customers |
++------+-----------------+
+| 1001 |               2 |
+| 1002 |               2 |
++------+-----------------+
+2 rows in set (0.00 sec)
+
+
+
+
+**************************************************************************************************************************************
+
+
+
+
+Que 27 ->Find salespeople with customers located in their own cities.
+
+Query  ->SELECT S.snum,S.sname,C.cname,S.city AS city 
+       ->FROM salespeople S, Customers C  
+       ->WHERE C.city = S.city;
++------+--------+----------+---------+
+| snum | sname  | cname    | city    |
++------+--------+----------+---------+
+| 1001 | Peel   | hoffman  | London  |
+| 1004 | Motika | hoffman  | London  |
+| 1008 | Fran   | hoffman  | London  |
+| 1002 | Serres | Liu      | SanJose |
+| 1001 | Peel   | Clemens  | London  |
+| 1004 | Motika | Clemens  | London  |
+| 1008 | Fran   | Clemens  | London  |
+| 1002 | Serres | Cisneros | SanJose |
++------+--------+----------+---------+
+8 rows in set (0.00 sec)
+
+
+
+
+**************************************************************************************************************************************
+
+
+
+
+Que 28 -> Find all salespeople whose name starts with ‘P’ and fourth character is ‘L’.
+
+Query  -> SELECT snum,sname  
+       -> FROM salespeople  
+       -> WHERE sname LIKE 'P__L';
++------+-------+
+| snum | sname |
++------+-------+
+| 1001 | Peel  |
++------+-------+
+1 row in set (0.00 sec)
+
+
+
+
+**************************************************************************************************************************************
+
+
+
+
+Que 29 -> Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number.
+
+
+Query -> SELECT onum,odate,amt 
+      -> FROM Orders 
+      -> WHERE Orders.cnum=(SELECT cnum
+                            FROM Customers              
+                            WHERE cname='Cisneros');
++------+------------+---------+
+| onum | odate      | amt     |
++------+------------+---------+
+| 3001 | 1990-03-10 |   18.69 |
+| 3006 | 1990-03-10 | 1098.16 |
++------+------------+---------+
+2 rows in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+Que 30 ->Find the largest orders for Serres and Rifkin.
+
+Query  ->SELECT snum,MAX(amt) AS Total_Amount 
```

Comment:
Display salesperson name here so we know which order is from Serres and which one is from Rifkin.

Reply:
Sure sir.